sklearn/ensemble/tests/test_partial_dependence.py

```diff
@@ -47,6 +48,24 @@ def test_partial_dependence_classifier():
     assert axes is None
     assert_array_equal(pdp, pdp_2)
 
+    # with trivial (no-op) sample weights
+    clf.fit(X, y, sample_weight=np.ones(len(y)))
+
+    pdp_w, axes_w = partial_dependence(clf, [0], X=X, grid_resolution=5)
+
+    assert pdp_w.shape == (1, 4)
+    assert axes_w[0].shape[0] == 4
+    assert_array_equal(pdp_w, pdp)
+
+    # with non-trivial sample weights
+    clf.fit(X, y, sample_weight=w)
+
+    pdp_w2, axes_w2 = partial_dependence(clf, [0], X=X, grid_resolution=5)
+
+    assert pdp_w2.shape == (1, 4)
+    assert axes_w2[0].shape[0] == 4
+    assert pdp_w2[0, 0] != pdp_w[0, 0]
```

Comment:
Instead of just checking the first one, maybe check the mean squared difference?